### PR TITLE
Added data filter feature

### PR DESCRIPTION
**Pull request by serenabenny**
[UIOMaticNameField]
        [UIOMaticFilterField]
        public string Name { get; set; }

image
I didn't know how to set the button text for display.

### CODE REVIEW

**Comment by serenabenny** [comment]
image

File: src/UIOMatic/App_Plugins/UIOMatic/backoffice/uioMaticTree/list.html [hunk 1]
```diff
@@ -19,9 +19,25 @@
...
+                            <select ng-model="property.OperatorID"
+                                    data-ng-options="operator.id as operator.name for operator in Operators"></select>
+                            <div ng-include="property.View"></div>
+                        </div>
+                        <!--div ng-include="property.View"></div-->
+                    </umb-control-group>
+
+                    <div class="btn-group" ng-show="isFilterForm">
+                        <a class="btn btn-info" ng-click="filter()" prevent-default="">
+                            <localize key="actions_filter">Query</localize>
```
Comment (serenabenny) [inline src/UIOMatic/App_Plugins/UIOMatic/backoffice/uioMaticTree/list.html]: @TimGeyssens Where to set button caption.

**Comment by TimGeyssens** [comment]
Nice, but not sure about the UI placement, seems to take up a lot of place

File: src/UIOMatic/App_Plugins/UIOMatic/backoffice/uioMaticTree/list.html (hunk 1 above)
Comment (TimGeyssens) [inline src/UIOMatic/App_Plugins/UIOMatic/backoffice/uioMaticTree/list.html]: @serenabenny just remove the localise element, and keep the inner text, we won't be translating the UI so no need for it :)

**Comment by serenabenny** [comment]
if not set [UIOMaticFilterField] Attribute,The UI same as before. that 's will auto hide.

**Comment by serenabenny** [comment]
I am more familiar with the back-end programming, The UI just copy from a part of edit.html . ::)

**Comment by TimGeyssens** [comment]
Ok :) might be better to open in the side dialog instead of the top placement

**Comment by serenabenny** [comment]
Nice,but I don't know how to do that. :(

**Comment by TimGeyssens** [comment]
@serenabenny take a look at these docs http://umbraco.github.io/Belle/#/api/umbraco.services.dialogService , basically you add it to it's own html file and add a new controller, then you use the dialog service to open that

**Comment by serenabenny** [comment]
image

**Comment by TimGeyssens** [comment]
Nice! Another awesome feature would be that you could also select wich field to query on (since it's now fixed to Name)

**Comment by serenabenny** [comment]
No , All field can query on which set to [UIOMaticFilterField] Attribute.

**Comment by TimGeyssens** [comment]
Ah so you can set the attribute to multiple properties on the same class? Mind sharing a screenshot of how the UI looks then when this is the case?

**Comment by serenabenny** [comment]
image
image

**Comment by TimGeyssens** [comment]
Nice!

**Comment by akeilox** [comment]
@serenabenny this is looking great!

If you specify multiple FilterField, like in the sshot, does it filter with AND query when both selected?

**Comment by serenabenny** [comment]
@akeilox Yes , that's AND query.

**Comment by TimGeyssens** [comment]
@serenabenny would be ace if the query builder would allow you to specify, so you can add more rules (and / or / not )

**Comment by TimGeyssens** [comment]
Aka something in the style of this http://wiki.sensenet.com/images/5/55/Querybuilder.png

**Comment by serenabenny** [comment]
@TimGeyssens I found Angular Query Builder  at https://github.com/mfauveau/angular-query-builder
,but I didn't know how to integrate . can you help?

**Comment by TimGeyssens** [comment]
Ow looks nice, sure, will start a new branch where we can work on it together

**Comment by serenabenny** [comment]
OK,Let's do it!

**Pull request closed without merging**